### PR TITLE
WIP: Add dropdown navigation menu

### PR DESCRIPTION
**Pull request by fanxhe-pair**

### CODE REVIEW

File: lib/livingstyleguide/document.rb [hunk 1]
```diff
@@ -12,6 +12,8 @@ class LivingStyleGuide::Document < ::Tilt::Template
   attr_accessor :syntax
   attr_reader :scope
 
+  @@navigation = []
```
Comment (hagenburger) [inline lib/livingstyleguide/document.rb]: What happens if you compile two LSG files within one process?

File: lib/livingstyleguide/document.rb [hunk 2]
```diff
@@ -29,6 +31,65 @@ class LivingStyleGuide::Document < ::Tilt::Template
       end
     end
   end
```
Comment (hagenburger) [inline lib/livingstyleguide/document.rb]: Empty line missing.

File: lib/livingstyleguide/document.rb [hunk 3]
```diff
@@ -29,6 +31,65 @@ class LivingStyleGuide::Document < ::Tilt::Template
...
+  end
+
+  def self.navigation
+    menu = '<nav class="lsg--navigation"><ul>'
+    index = 0
+    headline_closed         = true
+    sub_headline_closed     = true
+    sub_sub_headline_closed = true
+
+    while index < @@navigation.length
```
Comment (hagenburger) [inline lib/livingstyleguide/document.rb]: The following code is hard to read. Try splitting it up into smaller methods with self-explaining names.

File: stylesheets/livingstyleguide/_navigation.scss [hunk 4]
```diff
@@ -0,0 +1,24 @@
+nav.lsg--navigation {
```
Comment (hagenburger) [inline stylesheets/livingstyleguide/_navigation.scss]: No need to define `nav` in this selector.

File: stylesheets/livingstyleguide/_navigation.scss [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
```
Comment (hagenburger) [inline stylesheets/livingstyleguide/_navigation.scss]: No empty line here.

File: stylesheets/livingstyleguide/_navigation.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
+nav.lsg--navigation {
+  
+  left: 0px;
```
Comment (hagenburger) [inline stylesheets/livingstyleguide/_navigation.scss]: `0` in CSS does not need a unit.

File: stylesheets/livingstyleguide/_navigation.scss [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  
+  ul {
+    list-style: none;
+     
+    li:hover > ul {
+      display: block;
+    }
+     
+    li a {
+        color: $lsg--color;
```
Comment (hagenburger) [inline stylesheets/livingstyleguide/_navigation.scss]: Indentation level.

File: stylesheets/livingstyleguide/_navigation.scss [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+        font-size: $lsg--base-font-size;
+        font-style: $lsg--base-font;        
+    }
+      
+    ul{
```
Comment (hagenburger) [inline stylesheets/livingstyleguide/_navigation.scss]: Missing space.

File: lib/livingstyleguide/document.rb (hunk 1 above)
Comment (fanxhe-pair) [inline lib/livingstyleguide/document.rb]: We tested to import two files and just generates one navigation menu [here](https://github.com/fanxhe/livingstyleguide/blob/add-navigation-menu/lib/livingstyleguide/command_line_interface.rb#L13https://github.com/fanxhe/livingstyleguide/blob/add-navigation-menu/lib/livingstyleguide/command_line_interface.rb%23L13)

**Comment by fanxhe-pair** [comment]
@hagenburger we are working on your feedback, the only one question that stills remaining is: What do you think about the headline levels and the issue with sub-sub-headlines and the hover effect?

**Comment by anxhe** [comment]
@jhilden We have some work in progress in #179, it already handles sub-headlines and sub-sub-headlines, but it has a hover effect that misbehaves for sub-sub-headlines, it also needs refactor though @hagenburger

**Comment by hagenburger** [comment]
I’ll leave it closed until this gets stable. I’m totally fine with having this a little later than v2.0.0.

**Pull request closed without merging**